src/components/CategoryPicker/index.js

```diff
@@ -1,47 +1,94 @@
-import React, {useMemo} from 'react';
-import _ from 'underscore';
+import React, {useMemo, useState} from 'react';
 import {withOnyx} from 'react-native-onyx';
+import _ from 'underscore';
+import lodashGet from 'lodash/get';
 import ONYXKEYS from '../../ONYXKEYS';
 import {propTypes, defaultProps} from './categoryPickerPropTypes';
-import OptionsList from '../OptionsList';
 import styles from '../../styles/styles';
-import ScreenWrapper from '../ScreenWrapper';
 import Navigation from '../../libs/Navigation/Navigation';
 import ROUTES from '../../ROUTES';
+import CONST from '../../CONST';
+import * as IOU from '../../libs/actions/IOU';
+import * as OptionsListUtils from '../../libs/OptionsListUtils';
+import OptionsSelector from '../OptionsSelector';
+import useLocalize from '../../hooks/useLocalize';
+
+function CategoryPicker({policyCategories, reportID, policyID, iouType, iou, recentlyUsedPolicyCategories}) {
+    const {translate} = useLocalize();
+    const [searchValue, setSearchValue] = useState('');
+
+    const policyCategoriesAmount = _.size(policyCategories);
 
-function CategoryPicker({policyCategories, reportID, iouType}) {
-    const sections = useMemo(() => {
-        const categoryList = _.chain(policyCategories)
-            .values()
-            .map((category) => ({
-                text: category.name,
-                keyForList: category.name,
-                tooltipText: category.name,
-            }))
-            .value();
+    const selectedOptions = useMemo(() => {
+        if (!iou.category) {
+            return [];
+        }
 
         return [
             {
-                data: categoryList,
+                name: iou.category,
+                enabled: true,
+                accountID: null,
             },
         ];
-    }, [policyCategories]);
+    }, [iou.category]);
+
+    const initialFocusedIndex = useMemo(() => {
+        if (policyCategoriesAmount < CONST.CATEGORY_LIST_THRESHOLD && selectedOptions.length > 0) {
```

Comment:
NAB: You could make this a bit cleaner by assigning `policyCategoriesAmount < CONST.CATEGORY_LIST_THRESHOLD` to a variable and then you can use the negation of it instead of `shouldShowTextInput = policyCategoriesAmount >= CONST.CATEGORY_LIST_THRESHOLD` down below